h2o-algos/src/main/java/hex/anovaglm/AnovaGLMModel.java

```diff
@@ -0,0 +1,213 @@
+      return "Anova for Generalized Linear Model";
+    }
+
+    @Override
+    public String javaName() { return AnovaGLMModel.class.getName(); }
+
+    @Override
+    public long progressUnits() {
+      return 1;
+    }
+
+    public MissingValuesHandling missingValuesHandling() {
+      if (_missing_values_handling instanceof MissingValuesHandling)
+        return (MissingValuesHandling) _missing_values_handling;
+      assert _missing_values_handling instanceof DeepLearningModel.DeepLearningParameters.MissingValuesHandling;
+      switch ((DeepLearningModel.DeepLearningParameters.MissingValuesHandling) _missing_values_handling) {
+        case MeanImputation:
+          return MissingValuesHandling.MeanImputation;
+        case Skip:
+          return MissingValuesHandling.Skip;
+        default:
+          throw new IllegalStateException("Unsupported missing values handling value: " + _missing_values_handling);
+      }
+    }
+
+    public boolean imputeMissing() {
+      return missingValuesHandling() == MissingValuesHandling.MeanImputation ||
+              missingValuesHandling() == MissingValuesHandling.PlugValues;
+    }
+
+    public DataInfo.Imputer makeImputer() {
+      if (missingValuesHandling() == MissingValuesHandling.PlugValues) {
+        if (_plug_values == null || _plug_values.get() == null) {
+          throw new IllegalStateException("Plug values frame needs to be specified when Missing Value Handling = PlugValues.");
+        }
+        return new GLM.PlugValuesImputer(_plug_values.get());
+      } else { // mean/mode imputation and skip (even skip needs an imputer right now! PUBDEV-6809)
+        return new DataInfo.MeanImputer();
+      }
+    }
+  }
+
+  public static class AnovaGLMModelOutput extends Model.Output {
+    DataInfo _dinfo;
+    public long _training_time_ms;
+    public String[][] _coefficient_names; // coefficient names of all models
+    Family _family;
+    public String _transformed_columns_key;
+    public Key<Frame> _transformedColumnKey;
+    public TwoDimTable[] _coefficients_table;
+
+    @Override
+    public ModelCategory getModelCategory() {
+      switch (_family) {
+        case quasibinomial:
+        case fractionalbinomial:
+        case binomial: return ModelCategory.Binomial;
+        case multinomial: return ModelCategory.Multinomial;
+        case ordinal: return ModelCategory.Ordinal;
+        default: return ModelCategory.Regression;
+      }
+    }
+
+    public String[][] coefficientNames() { return _coefficient_names; }
+
+    public AnovaGLMModelOutput(AnovaGLM b, DataInfo dinfo) {
+      super(b, dinfo._adaptedFrame);
+      _dinfo = dinfo;
+      _domains = dinfo._adaptedFrame.domains();
+      _family = b._parms._family;
+    }
+  }
+
+  /**
+   * The Type III SS calculation, degree of freedom, F-statistics and p-values will be included in the model
+   * summary.  For details on how those are calculated, refer to AnovaGLMTutorial section V.
```

Comment:
It would be nice to have the AnovaGLMTurorial saved in a permanent link and post it here. Or in the header comment of this class.

Reply:
Veronika: Great suggestion.  I am working on this one.  Will add it in once I get it.